4-WebApp-your-API/4-2-B2C/TodoListService/AuthorizationPolicies/OperationScopeHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Infrastructure;
+using Microsoft.Identity.Web;
+using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace TodoListService.AuthorizationPolicies
+{
+    /// <summary>
+    /// AuthorizationHandler that will check if the scope claim has the requirement value
+    /// </summary>
+    public class OperationScopeHandler : AuthorizationHandler<OperationAuthorizationRequirement>
```

Comment:
Would we be able to specialize `OperationAuthorizationRequirement` to `ScopeRequirement` and have a property named `Scope` ?